pyccl/halos/hmfunc/press74.py

```diff
@@ -11,34 +12,25 @@ class MassFuncPress74(MassFunc):
     This parametrization is only valid for 'fof' masses.
 
     Args:
-        cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
         mass_def (:class:`~pyccl.halos.massdef.MassDef`):
             a mass definition object.
             this parametrization accepts FoF masses only.
-            If `None`, FoF masses will be used.
+            The default is 'fof'.
         mass_def_strict (bool): if False, consistency of the mass
             definition will be ignored.
     """
     name = 'Press74'
 
-    def __init__(self, cosmo, mass_def=None, mass_def_strict=True):
-        super(MassFuncPress74, self).__init__(cosmo,
-                                              mass_def,
-                                              mass_def_strict)
+    @warn_api
+    def __init__(self, *,
+                 mass_def=MassDef('fof', 'matter'),
+                 mass_def_strict=True):
+        super().__init__(mass_def=mass_def, mass_def_strict=mass_def_strict)
 
-    def _default_mdef(self):
-        self.mdef = MassDef('fof', 'matter')
-
-    def _setup(self, cosmo):
-        self.norm = np.sqrt(2/np.pi)
-
-    def _check_mdef_strict(self, mdef):
-        if mdef.Delta != 'fof':
-            return True
-        return False
+    def _check_mass_def_strict(self, mass_def):
+        return mass_def.Delta != "fof"
 
     def _get_fsigma(self, cosmo, sigM, a, lnM):
         delta_c = 1.68647
-
         nu = delta_c/sigM
-        return self.norm * nu * np.exp(-0.5 * nu**2)
+        return np.sqrt(2/np.pi) * nu * np.exp(-0.5 * nu**2)
```

Comment:
You claim that np.sqrt(2/np.pi) only gets run once by the interpreter, but I think it happens every time you run this function. Convince me otherwise.

Reply:
I didn't claim that, this is only the case with numbers, and things that can be compiled (not evaluated in greneral) to bytecode. So, `(2/3.14)**0.5` would be cached when the interpreter first runs the program, but `np.sqrt(2/np.pi)` would not.

In this case, I brought it down there because it enhances code legibility. Execution time for this is ~300 ns so completely negligible.

Reply:
Generally, in fast languages such as C/C++, it makes sense to define these `_one_over_4_pi` variables for code that is run multiple times. In Python, it makes no difference wrt to runtime. And since it reads better this way, let's go for readability at least.

Reply:
So this is less efficient than precalculating it, in which case, precalculate it. If you are worried about code readability, add a comment on what the constant is.

Reply:
Less efficient by less than a microsecond, yes. If we care so much about microseconds we should probably switch to C++...
Anyway, I precalculate it now.